src/oscar/apps/catalogue/product_attributes.py

```diff
@@ -13,25 +13,30 @@ class ProductAttributesContainer(object):
 
     def __setstate__(self, state):
         self.__dict__ = state
-        self.initialised = False
+        self._initialised = False
 
     def __init__(self, product):
-        self.product = product
-        self.initialised = False
+        self._product = product
+        self._initialised = False
 
     def initiate_attributes(self):
         values = self.get_values().select_related('attribute')
+        self._initialised = True
```

Comment:
You get an infinite loop if this is set *after* the loop below, because that itself is using `setattr`...